### PR TITLE
Feat device file exists function

### PR DESCRIPTION
**Pull request by lohanidamodar**
Adding new function in device to check if the file exists in the given path.

### CODE REVIEW

**Comment by eldadfux** [comment]
I think new methods are missing tests

**Comment by lohanidamodar** [comment]
Will add them!

File: src/Storage/Device/S3.php [hunk 1]
```diff
@@ -386,9 +402,9 @@ private function resetResponse(): void
 
     /**
      * Get file info
-     * @return Array | false
+     * @return array | false
```
Comment (lohanidamodar) [inline src/Storage/Device/S3.php:405]: @eldadfux is it a good idea, when there is error, returning false, if yes how to set return type of this function?

Also, I am think we need to introduce exceptions as well in this library, right now for AWS adapter, I am just returning false, anywhere there's error. This makes a lot difficult while debugging errors.

Comment (eldadfux) [reply src/Storage/Device/S3.php:405]: Yes. You can throw the default PHP exception for all those cases, and keep consistent return type for better type strict codebase. If we'll need more types, we can later extend with our own custom exceptions.

Comment (lohanidamodar) [reply src/Storage/Device/S3.php:405]: @eldadfux this is not related to this feature, so can we merge this and i'll introduce another fix PR?

**Pull request merged**